### PR TITLE
fix(bug): sync from da and p2p when starting a node

### LINKED ISSUE #771: nodes are not syncing from p2p, while still syncing from DA
nodes disregard any received blocks from p2p, while they are trying to sync from DA on startup, which should not happen.

### PR DESCRIPTION
**Pull request by srene**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close #771 , https://github.com/dymensionxyz/dymint/issues/761

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Review by danwt — APPROVED** [review]

File: block/retriever.go [hunk 1]
```diff
@@ -44,33 +43,33 @@ func (m *Manager) syncUntilTarget(syncTarget uint64) error {
...
-	m.logger.Debug("Sync until target: updated state index pre loop", "stateIndex", stateIndex, "height", h, "syncTarget", syncTarget)
-
 	for currentHeight < syncTarget {
-		currStateIdx := atomic.LoadUint64(&m.LastState.SLStateIndex) + 1
-		m.logger.Info("Syncing until target", "height", currentHeight, "state_index", currStateIdx, "syncTarget", syncTarget)
-		settlementBatch, err := m.SLClient.RetrieveBatch(currStateIdx)
+		var stateIndex uint64
+		h := m.Store.Height() + 1
+
+		m.logger.Debug("Sync until target: updated state index pre loop", "height", h, "syncTarget", syncTarget)
```
Comment (danwt) [inline block/retriever.go:50]: this log is not pre loop anymore, and might be too noisy?

**Review by danwt — CHANGES_REQUESTED** [review]
We need to delete the state index field

https://github.com/dymensionxyz/dymint/blob/df192a9904800b7b1a8bf60a17987cd6630fe5f7/types/state.go#L41

**Review by danwt — DISMISSED** [review]

File: block/manager.go [hunk 2]
```diff
@@ -252,8 +252,6 @@ func (m *Manager) onNewGossipedBlock(event pubsub.Message) {
 		m.logger.Debug("caching block", "block height", block.Header.Height, "store height", m.Store.Height())
 	}
 
-	m.retrieverMutex.Unlock() // have to give this up as it's locked again in attempt apply, and we're not re-entrant
```
Comment (danwt) [inline block/manager.go:255]: ditto

File: da/celestia/config.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ type Config struct {
 var CelestiaDefaultConfig = Config{
 	BaseURL:        "http://127.0.0.1:26658",
 	AppNodeURL:     "",
-	Timeout:        30 * time.Second,
+	Timeout:        5 * time.Second,
```
Comment (danwt) [inline da/celestia/config.go:45]: double checking, was this intentional?

File: block/manager.go [hunk 4]
```diff
@@ -237,8 +239,7 @@ func (m *Manager) onNodeHealthStatus(event pubsub.Message) {
 // onNewGossippedBlock will take a block and apply it
 func (m *Manager) onNewGossipedBlock(event pubsub.Message) {
 	m.retrieverMutex.Lock() // needed to protect blockCache access
-
-	m.logger.Debug("Received new block via gossip", "n cachedBlocks", len(m.blockCache))
+	m.logger.Debug("Received new block event", "eventData", event.Data(), "cachedBlocks", len(m.blockCache))
```
Comment (danwt) [inline block/manager.go:242]: this was changed wrongly

**Review by danwt — CHANGES_REQUESTED** [review]

File: block/manager.go (hunk 4 above)
Comment (srene) [reply block/manager.go]: fixed

**Review by danwt — APPROVED** [review]

**Review by danwt — DISMISSED** [review]

**Review by danwt — APPROVED** [review]

**Review by omritoptix — APPROVED** [review]

**Review by danwt — DISMISSED** [review]

**Pull request merged**